### PR TITLE
Fix issues related to delete button in name value pair editor

### PR DESCRIPTION
**Pull request by gijohn**
Fixes: https://jira.coreos.com/browse/ODC-1481

delete-button

1. Show delete button only when there is a value or multiple rows
2. Dragging the rows creates multiple duplicate items - Fixed.
3. Console warning logs - Fixed

### CODE REVIEW

File: frontend/public/components/utils/name-value-editor.jsx [hunk 1]
```diff
@@ -22,9 +22,9 @@ export const NameValueEditor = withDragDropContext(class NameValueEditor extends
   }
 
   _append() {
-    const {updateParentData, nameValuePairs, nameValueId, allowSorting} = this.props;
+    const {updateParentData, nameValuePairs, nameValueId} = this.props;
 
-    updateParentData({nameValuePairs: allowSorting ? nameValuePairs.concat([['', '', nameValuePairs.length]]) : nameValuePairs.concat([['', '']])}, nameValueId);
+    updateParentData({nameValuePairs: nameValuePairs.concat([['', '', nameValuePairs.length]])}, nameValueId);
```
Comment (gijohn) [inline frontend/public/components/utils/name-value-editor.jsx]: index is used as key. Passing empty is causing console warning logs.

File: frontend/public/components/utils/name-value-editor.jsx [hunk 2]
```diff
@@ -37,8 +37,9 @@ export const NameValueEditor = withDragDropContext(class NameValueEditor extends
     const {updateParentData, nameValueId} = this.props;
     const nameValuePairs = _.cloneDeep(this.props.nameValuePairs);
     nameValuePairs.splice(i, 1);
+    nameValuePairs.forEach((values, i) => values[2] = i); // update the indices in order.
 
-    updateParentData({nameValuePairs: nameValuePairs.length ? nameValuePairs : [['', '']]}, nameValueId);
+    updateParentData({nameValuePairs: nameValuePairs.length ? nameValuePairs : [['', '', 0]]}, nameValueId);
```
Comment (gijohn) [inline frontend/public/components/utils/name-value-editor.jsx]: index is used as key. Passing empty is causing console warning logs.

**Comment by gijohn** [comment]
@serenamarie125 Please check if the fix is good. Instead of hiding the button we can disable too.

**Comment by christianvogt** [comment]
@gijohn the issue has been discussed already and the behavior of the delete button is still pending UX confirmation. Please remove that change from your PR. The other problems should have been raised as a separate issue.

**Comment by gijohn** [comment]
Closing this PR. Will create a separate PR to fix other issues

**Pull request closed without merging**